components/director/internal/open_resource_discovery/validation.go

```diff
@@ -685,6 +693,16 @@ func isValidDate(date *string) validation.RuleFunc {
 	}
 }
 
+func isAPIPartOfBundles(partOfConsumptionBundles []*model.ConsumptionBundleReference) validation.RuleFunc {
```

Comment:
The name `isAPIPartOfBundles` does not imply which is true and which is false.

This is fine when used as a standalone function, however it is used in validation like:

```
validation.When(api.TargetURLs == nil, validation.By(isAPIPartOfBundles(api.PartOfConsumptionBundles))
```

When you read this line you do not know that it is valid when it is part of consumption bundle or when it is not.

I would suggest something like `notPartOfConsumptionBundles`